### PR TITLE
[prometheus-node-exporter] Add externalTrafficPolicy to prometheus node exporter service spec

### PR DESCRIPTION
**Pull request by junchil**
#### What this PR does / why we need it

Add externalTrafficPolicy config option to node exporter service.

When the service type set to nodeport, we can have this externalTrafficPolicy config set to local so service only forward request to the local node exporter pod.

@gianrubio @zanhsieh @zeritti 

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-node-exporter/templates/service.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@ spec:
 {{- if .Values.service.ipDualStack.enabled }}
   ipFamilies: {{ toYaml .Values.service.ipDualStack.ipFamilies | nindent 4 }}
   ipFamilyPolicy: {{ .Values.service.ipDualStack.ipFamilyPolicy }}
+  externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy }}
 {{- end }}
```
Comment (zeritti) [inline charts/prometheus-node-exporter/templates/service.yaml:18]: The field should not be affected by the conditional on dual stack.

I'd suggest to consider the value only if set by the user, i.e. let us keep it unset by default. 

```suggestion
{{- end }}
{{- if .Values.service.externalTrafficPolicy }}
  externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy }}
{{- end }}
```

File: charts/prometheus-node-exporter/values.yaml [hunk 2]
```diff
@@ -98,6 +98,7 @@ service:
     enabled: false
     ipFamilies: ["IPv6", "IPv4"]
     ipFamilyPolicy: "PreferDualStack"
+  externalTrafficPolicy: Cluster
```
Comment (zeritti) [inline charts/prometheus-node-exporter/values.yaml:101]:
```suggestion
  externalTrafficPolicy: ""
```

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @junchil, for your PR. Please, see my comments below.

File: charts/prometheus-node-exporter/Chart.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ keywords:
   - prometheus
   - exporter
 type: application
-version: 4.33.0
+version: 4.33.1
```
Comment (zeritti) [inline charts/prometheus-node-exporter/Chart.yaml:9]: A new feature should raise the minor number.

```suggestion
version: 4.34.0
```

**Review by zeritti — CHANGES_REQUESTED** [review]

**Comment by junchil** [comment]
Hi @zeritti thank you for your review. pls review it again, thanks!

**Review by zeritti — APPROVED** [review]

**Comment by junchil** [comment]
Hi @zeritti how can I merge this pr?

**Pull request merged**